Source/core/SharedBuffer.h

```diff
@@ -65,34 +65,6 @@ namespace Core {
         SharedBuffer& operator=(const SharedBuffer&) = delete;
 
     private:
-        class Semaphore {
-        private:
-            Semaphore() = delete;
-            Semaphore(const Semaphore&) = delete;
-            Semaphore& operator=(const Semaphore&) = delete;
-
-        public:
-#ifdef __WINDOWS__
-            Semaphore(const TCHAR name[]);
-#else
-            Semaphore(sem_t* storage);
-            //Semaphore(sem_t* storage, bool initialize) {
-            //}
-#endif
-            ~Semaphore();
-
-        public:
-            uint32_t Lock(const uint32_t waitTime);
-            uint32_t Unlock();
-            bool IsLocked();
-
-        private:
-#ifdef __WINDOWS__
-            HANDLE _semaphore;
-#else
-            sem_t* _semaphore;
-#endif
-        };
         struct Administration {
 
             uint32_t _bytesWritten;
```

Comment:
Now we did our best to hide the imeplementation details as much as possible, however, here we do refer to the hidden details. Can we not create something like:
 ```
 #ifndef __WINDOWS__
            uint8_t _producerSpace[SharedSemapore::Size()];
            uint8_t _consumerSpace[SharedSemapore::Size()];
#endif
       ```

Reply:
Both sem_t* and HANDLE can be hidden behind void*. Again no purpose on Size() on Windows.